server/src/test/java/org/candlepin/dto/AbstractDTOTest.java

```diff
@@ -88,6 +89,18 @@ public AbstractDTOTest(Class<T> dtoClass) {
         this.copyConstructor = constructor;
     }
 
+    /**
+     * Method to skip testing any field in any DTO class by
+     * not including their accessor method for testing.
+     * Defaults to false
+     *
+     * @param methodName getter method of field.
+     * @return boolean value
+     */
+    protected boolean skipMethod(String methodName) {
```

Comment:
This shouldn't be a thing. If a field isn't implemented or otherwise won't be valuable, we should be verifying that it always returns null or, as is the task here, determining whether or not the model is valid.

Reply:
Actually AbstractDTOTest is a generic class which perform test for all fields in inheritance hierarchy, for ProductCertificateDTO.class which now override the Getter/Setter method for `CertificateSerialDTO aka serial` to return null for Get method and do nothing for Set method. This lead to failure of few Junit due to assert calls on DTO for test like EqualsForIndividualFields. I can bypass this by adding a dummy value for serial in ProductCeritficateDTOTest, but ProductCertificateDTO has nothing to do with Serial CertificateSerialDTO and doesn't make sense.

Reply:
The point here is that if we have a method, we should be testing it -- especially when the work for that test is literally just adding a one-liner to set the expected values.

Moreover, you're touching on the reason the task exists: the model may be incorrect here and it might be worth either completely separating them or refactoring them and making them more connected. This could occur on the entity side, the DTO side, both or neither. Perhaps some of the details of the task were lost in the conversion from Trello to Jira, but that's the objective here. This was supposed to be an investigative task first and foremost.